projects/plugins/wpcomsh/connection/class-atomic-storage-provider.php

```diff
@@ -47,11 +49,19 @@ public function get( $option_name ) {
 
 			switch ( $option_name ) {
 				case 'blog_token':
-					return $persistent_data->JETPACK_BLOG_TOKEN; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+					return empty( $persistent_data->JETPACK_BLOG_TOKEN ) ? false : $persistent_data->JETPACK_BLOG_TOKEN; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
 
 				case 'id':
-					$blog_id = $persistent_data->JETPACK_BLOG_ID; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
-					return $blog_id ? intval( $blog_id ) : null;
+					return intval( $persistent_data->JETPACK_BLOG_ID ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+				case 'master_user':
+					$email = $persistent_data->JETPACK_CONNECTION_OWNER_EMAIL; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+					return $email ? $this->get_master_user_id( $email ) : false;
```

Comment:
Nitpick: since `get_master_user_id()` also checks if the email is empty and valid, and returns `false` otherwise, we can probably simplify this a bit:
```suggestion
					return $this->get_master_user_id( $email );
```

Reply:
It looks a bit different now, but if the `get_master_user_id()` returns false, we need to return `null`.